directory/directory_dest.go

```diff
@@ -147,3 +183,40 @@ func (d *dirImageDestination) PutSignatures(signatures [][]byte) error {
 func (d *dirImageDestination) Commit() error {
 	return nil
 }
+
+// returns true if path exists
+func dirExists(path string) bool {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true
+	}
+	return false
+}
+
+// returns true if directory is empty
+func isDirEmpty(path string) (bool, error) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return false, err
+	}
+	if len(files) > 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+// deletes the contents of a directory
+func removeDirContents(path string) error {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return err
+	}
+
+	for _, file := range files {
+		err := os.Remove(filepath.Join(path, file.Name()))
```

Comment:
Should use os.RemoveAll, since this could be a directory.

Reply:
os.RemoveAll will remove the parent directory also. I just want to remove the contents of the directory and os.Remove removes both files and directories.

Reply:
I don't believe os.Remove will remove a non empty subdir.  I am suggesting you use RemoveAll on the contents of the directory, not the full directory.

Reply:
you are right, I misunderstood the docs on os.Remove. Fixed